source/containers.tex

```diff
@@ -14292,6 +14384,1710 @@
+    flat_map(sorted_unique_t, KeyContainer, MappedContainer, Allocator)
+      -> flat_map<typename KeyContainer::value_type, typename MappedContainer::value_type,
+                  less<typename KeyContainer::value_type>, KeyContainer, MappedContainer>;
+
+  template<class InputIterator, class Compare = less<@\exposid{iter-key-type}@<InputIterator>>>
+    flat_map(InputIterator, InputIterator, Compare = Compare())
+      -> flat_map<@\exposid{iter-key-type}@<InputIterator>, @\exposid{iter-mapped-type}@<InputIterator>, Compare>;
+
+  template<class InputIterator, class Compare = less<@\exposid{iter-key-type}@<InputIterator>>>
+    flat_map(sorted_unique_t, InputIterator, InputIterator, Compare = Compare())
+      -> flat_map<@\exposid{iter-key-type}@<InputIterator>, @\exposid{iter-mapped-type}@<InputIterator>, Compare>;
+
+  template<ranges::input_range R, class Compare = less<@\exposid{range-key-type}@<R>>,
+           class Allocator>
+    flat_map(from_range_t, R&&, Compare = Compare(), Allocator = Allocator())
+      -> flat_map<@\exposid{range-key-type}@<R>, @\exposid{range-mapped-type}@<R>, Compare>;
+
+  template<ranges::input_range R, class Allocator>
+    flat_map(from_range_t, R&&, Allocator)
+      -> flat_map<@\exposid{range-key-type}@<R>, @\exposid{range-mapped-type}@<R>>;
+
+  template<class Key, class T, class Compare = less<Key>>
+    flat_map(initializer_list<pair<Key, T>>, Compare = Compare())
+      -> flat_map<Key, T, Compare>;
+
+  template<class Key, class T, class Compare = less<Key>>
+    flat_map(sorted_unique_t, initializer_list<pair<Key, T>>, Compare = Compare())
+        -> flat_map<Key, T, Compare>;
+
+  template<class Key, class T, class Compare, class KeyContainer, class MappedContainer,
+            class Allocator>
+    struct uses_allocator<flat_map<Key, T, Compare, KeyContainer, MappedContainer>, Allocator>
+      : bool_constant<uses_allocator_v<KeyContainer, Allocator> &&
+                      uses_allocator_v<MappedContainer, Allocator>> { };
+}
+\end{codeblock}
+
+\pnum
+The member type \tcode{containers} has the data members and special members
+specified above.
+It has no base classes or members other than those specified.
+
+\rSec3[flatmap.cons]{Constructors}
+
+\indexlibraryctor{flatmap}%
+\begin{itemdecl}
+flat_map(key_container_type key_cont, mapped_container_type mapped_cont);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Initializes \tcode{c.keys} with \tcode{std::move(key_cont)} and
+\tcode{c.values} with \tcode{std::move(mapped_cont)};
+value-initializes \tcode{compare};
+sorts the range \range{begin()}{end()} with respect to \tcode{value_comp()}; and
+finally erases the duplicate elements as if by:
+\begin{codeblock}
+auto zv = ranges::zip_view(c.keys, c.values);
+auto it = ranges::unique(zv, key_equiv(compare)).begin();
+auto dist = distance(zv.begin(), it);
+c.keys.erase(c.keys.begin() + dist, c.keys.end());
+c.values.erase(c.values.begin() + dist, c.values.end());
+\end{codeblock}
```

Comment:
```suggestion
finally erases the duplicate elements as if by \tcode{\exposid{erase-duplicate-elements}()}.
```

Reply:
Where do these expos-algorithms come from? Are you proposing we define them?

Reply:
Yes: https://github.com/cplusplus/draft/pull/5626#discussion_r922827185.

Reply:
Ah yes, I see. OK, let's do that later, but it is a good idea!